ConsentViewController/Classes/SourcePointClient/SourcepointClientCoordinator.swift

```diff
@@ -699,25 +709,33 @@ class SourcepointClientCoordinator: SPClientCoordinator {
         ) { handler($0) }
     }
 
+    func handleGPDRPostChoice(
+        _ action: SPAction,
+        _ getResponse: ChoiceAllResponse?,
+        _ postResponse: GDPRChoiceResponse
+    ) {
+        if action.type == .SaveAndExit {
+            state.gdpr?.tcfData = postResponse.TCData
+        }
+        state.gdpr?.uuid = postResponse.uuid
+        state.gdpr?.dateCreated = postResponse.dateCreated
+        state.gdpr?.expirationDate = postResponse.expirationDate
+        state.gdpr?.consentStatus = postResponse.consentStatus ?? getResponse?.gdpr?.consentStatus ?? ConsentStatus()
+        state.gdpr?.euconsent = postResponse.euconsent ?? getResponse?.gdpr?.euconsent ?? ""
+        state.gdpr?.vendorGrants = postResponse.grants ?? getResponse?.gdpr?.grants ?? SPGDPRVendorGrants()
+        state.gdpr?.webConsentPayload = postResponse.webConsentPayload ?? getResponse?.gdpr?.webConsentPayload
+        state.gdpr?.legIntCategories = postResponse.legIntCategories ?? getResponse?.gdpr?.legIntCategories
+        state.gdpr?.legIntVendors = postResponse.legIntVendors ?? getResponse?.gdpr?.legIntVendors
+        state.gdpr?.vendors = postResponse.vendors ?? getResponse?.gdpr?.vendors
+        state.gdpr?.categories = postResponse.categories ?? getResponse?.gdpr?.categories
+        storage.spState = state
+    }
+
     func reportGDPRAction(_ action: SPAction, _ getResponse: ChoiceAllResponse?, _ handler: @escaping ActionHandler) {
         postChoice(action, postPayloadFromGetCall: getResponse?.gdpr?.postPayload) { postResult in
             switch postResult {
                 case .success(let response):
-                    if action.type == .SaveAndExit {
-                        self.state.gdpr?.tcfData = response.TCData
-                    }
-                    self.state.gdpr?.uuid = response.uuid
-                    self.state.gdpr?.dateCreated = response.dateCreated
-                    self.state.gdpr?.consentStatus = response.consentStatus ?? getResponse?.gdpr?.consentStatus ?? ConsentStatus()
-                    self.state.gdpr?.euconsent = response.euconsent ?? getResponse?.gdpr?.euconsent ?? ""
-                    self.state.gdpr?.vendorGrants = response.grants ?? getResponse?.gdpr?.grants ?? SPGDPRVendorGrants()
-                    self.state.gdpr?.webConsentPayload = response.webConsentPayload ?? getResponse?.gdpr?.webConsentPayload
-                    self.state.gdpr?.legIntCategories = response.legIntCategories ?? getResponse?.gdpr?.legIntCategories
-                    self.state.gdpr?.legIntVendors = response.legIntVendors ?? getResponse?.gdpr?.legIntVendors
-                    self.state.gdpr?.vendors = response.vendors ?? getResponse?.gdpr?.vendors
-                    self.state.gdpr?.categories = response.categories ?? getResponse?.gdpr?.categories
-                    self.storage.spState = self.state
-
+                    self.handleGPDRPostChoice(action, getResponse, response)
```

Comment:
Moved the code above to new method `handleGDPRPostChoice`, for better readability.